cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -231,3 +239,18 @@ func isServiceActive(name string) (bool, error) {
 	}
 	return initSystem.ServiceIsActive(name), nil
 }
+
+// MutateCgroupDriver can be called to set the KubeletConfiguration cgroup driver to systemd
+// Currently this cannot be as part of Default() because the function is called for
+// upgrades too, which can break existing nodes after a kubelet restart.
+func MutateCgroupDriver(cfg *kubeadmapi.ClusterConfiguration) {
+	kubeletConfig := cfg.ComponentConfigs[KubeletGroup]
```

Comment:
Should we check for cfg.ComponentConfigs[KubeletGroup] being nil/not existing?

Reply:
i think currently the internal configuration is defaulted with blank component configs, but we can have a check + warning if this somehow fails.